### ISSUE TITLE
[Bug] Activity logger does not serialize Error objects

### ISSUE THREAD

**Issue by shawnjones253** [temporalio/sdk-typescript#1282]
### What are you really trying to do?

From an activity:
```
Context.current().log.error('Kablamo!', new Error('THIS IS AN ERROR!!!'))
```

### Describe the bug

```
Context.current().log.error('Kablamo!', new Error('THIS IS AN ERROR!!!'))
```
produces:
```
2023-11-08T02:17:00.169Z [ERROR] Kablamo! {
  isLocal: false,
  attempt: 10,
  namespace: 'default',
  taskToken: '...',
  workflowId: 'do-stuff',
  workflowRunId: '...,
  workflowType: 'doStuffWorkflow',
  activityId: '1',
  activityType: 'doStuffActivity',
  taskQueue: 'my-queue'
}
```

The log should contain an error object with at least the stack trace. This works for uncaught errors like:
```
throw new Error('THIS IS AN ERROR!!!')
```

produces:
```
2023-11-08T02:17:00.170Z [WARN] Activity failed {
  isLocal: false,
  attempt: 10,
  namespace: 'default',
  taskToken: '...',
  workflowId: 'do-stuff',
  workflowRunId: '...',
  workflowType: 'doStuffWorkflow',
  activityId: '1',
  activityType: 'doStuffActivity',
  taskQueue: 'my-queue',
  error: Error: THIS IS AN ERROR!!!
      at ...stackTrace...,
  durationMs: 2
}
```

### Environment/Versions

- OS and processor: M1 Mac
- Temporal Version: "1.8.1"
- Are you using Docker or Kubernetes or building Temporal from source? Docker temporalio/auto-setup

**Comment by mjameswh** [comment]
The second argument of `log.error` is a metadata object, not an error.

This works correctly:
```
Context.current().log.error('Kablamo!', { error: new Error('THIS IS AN ERROR!!!') })
```

**Comment by mjameswh** [comment]
> **Environment/Versions**
> - Temporal Version: "1.8.1"

By the way, some significant bugs were fixed since TS SDK 1.8.1. You should really upgrade to 1.8.6.
